Multicast e2e failed with error message "slice bounds out of range"

**Describe the bug**
Running multicast e2e tests(TestMulticast/testMulticastWithNoEncap/testMulticastBetweenPodsInThreeNodes/testMulticastMultipleReceiversTrafficToExternal) will report test failure, with the error message from Antrea Agent logs.
```
panic: runtime error: slice bounds out of range [:16] with capacity 4

goroutine 418 [running]:
antrea.io/libOpenflow/protocol.(*IGMPv3Query).MarshalBinary(0xc000845700)
        /antrea/vendor/antrea.io/libOpenflow/protocol/igmp.go:171 +0x3ef
antrea.io/antrea/pkg/ovs/openflow.(*ofPacketOutBuilder).igmpHeaderChecksum(0x6?)
        /antrea/pkg/ovs/openflow/ofctrl_packetout.go:505 +0x2d
antrea.io/antrea/pkg/ovs/openflow.(*ofPacketOutBuilder).Done(0xc000bec8b8)
        /antrea/pkg/ovs/openflow/ofctrl_packetout.go:403 +0x538
antrea.io/antrea/pkg/agent/openflow.(*client).SendIGMPQueryPacketOut(0xc0001de500, {0xc00005c678, 0x6, 0x203000?}, {0xc00019b56c, 0x4, 0x4}, 0xa?, {0x29a3350, 0xc000845700})
        /antrea/pkg/agent/openflow/client.go:1355 +0x26d
antrea.io/antrea/pkg/agent/multicast.(*IGMPSnooper).queryIGMP(0xc000317b90, {0xc000bd7cc4, 0x4, 0x4}, {0x3c5b880, 0x1, 0x0?})
        /antrea/pkg/agent/multicast/mcast_discovery.go:112 +0x123
antrea.io/antrea/pkg/agent/multicast.(*Controller).checkLastMember(0xc00090c900, {0xc000bd7cc4, 0x4, 0x4})
        /antrea/pkg/agent/multicast/mcast_controller.go:176 +0x67
antrea.io/antrea/pkg/agent/multicast.(*Controller).updateGroupMemberStatus(0xc00090c900, {0x2050780?, 0xc00030f8b0}, 0xc000c705a0)
        /antrea/pkg/agent/multicast/mcast_controller.go:163 +0x5a5
antrea.io/antrea/pkg/agent/multicast.(*Controller).addOrUpdateGroupEvent(0xc00090c900, 0xc000c705a0)
        /antrea/pkg/agent/multicast/mcast_controller.go:612 +0xa5
antrea.io/antrea/pkg/agent/multicast.(*Controller).eventHandler(0xc00090c900, 0xc00081a240)
        /antrea/pkg/agent/multicast/mcast_controller.go:101 +0xd9
created by antrea.io/antrea/pkg/agent/multicast.(*Controller).Run
        /antrea/pkg/agent/multicast/mcast_controller.go:375 +0x339
```


**To Reproduce**
<!-- Steps to reproduce the behavior. -->
Run the multicast e2e tests
**Expected**
Tests passed with no error

**Actual behavior**
Tests failed with the error described above

**Versions:**
<!--
Please provide the following information:
 - Antrea version (Docker image tag).
 - Kubernetes version (use `kubectl version`). If your Kubernetes components have [different versions](https://kubernetes.io/docs/setup/release/version-skew-policy/), please provide the version for all of them.
 - Container runtime: which runtime are you using (e.g. containerd, cri-o, docker) and which version are you using?
 - Linux kernel version on the Kubernetes Nodes (`uname -r`).
 - If you chose to compile the Open vSwitch kernel module manually instead of using the kernel module built into the Linux kernel, which version of the OVS kernel module are you using? Include the output of `modinfo openvswitch` for the Kubernetes Nodes.
-->
Antrea: 1.11(commit 12e622a791bbbd570e20a3a4784f8ed19f557e6c)
Kernel: 4.15.0-213-generic
k8s Version: v1.27.2
**Additional context**
<!-- Add any other context about the problem here, such as Antrea logs, kubelet logs, etc. -->

<!-- (Please consider pasting long output into a [GitHub gist](https://gist.github.com) or any other pastebin.) -->

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module antrea.io/antrea
 go 1.19
 
 require (
-	antrea.io/libOpenflow v0.12.0
+	antrea.io/libOpenflow v0.12.1
 	antrea.io/ofnet v0.9.0
 	github.com/ClickHouse/clickhouse-go/v2 v2.6.1
 	github.com/DATA-DOG/go-sqlmock v1.5.0

